objects/src/notes/metadata.rs

```diff
@@ -122,54 +161,125 @@ impl TryFrom<Word> for NoteMetadata {
 
 impl Serializable for NoteMetadata {
     fn write_into<W: ByteWriter>(&self, target: &mut W) {
-        self.sender.write_into(target);
-        target.write_u64(merge_type_and_hint(self.note_type, self.execution_hint));
-        self.tag.write_into(target);
-        self.aux.write_into(target);
+        // TODO: Do we need a serialization format that is different from the Word encoding? It was
+        // previously different.
+        Word::from(self).write_into(target);
     }
 }
 
 impl Deserializable for NoteMetadata {
     fn read_from<R: ByteReader>(source: &mut R) -> Result<Self, DeserializationError> {
-        let sender = AccountId::read_from(source)?;
-        let (note_type, note_execution_hint) = unmerge_type_and_hint(source.read_u64()?)
-            .map_err(|err| DeserializationError::InvalidValue(err.to_string()))?;
-        let tag = NoteTag::read_from(source)?;
-        let aux = Felt::read_from(source)?;
-
-        Self::new(sender, note_type, tag, note_execution_hint, aux)
-            .map_err(|err| DeserializationError::InvalidValue(err.to_string()))
+        let word = Word::read_from(source)?;
+        Self::try_from(word).map_err(|err| DeserializationError::InvalidValue(err.to_string()))
     }
 }
 
 // HELPER FUNCTIONS
 // ================================================================================================
 
-/// Encodes `note_type` and `note_execution_hint` into a [u64] such that the resulting number has
-/// the following structure (from most significant bit to the least significant bit):
+/// Merges the second felt of an [`AccountId`], a [`NoteType`] and the tag of a
+/// [`NoteExecutionHint`] into a single [`Felt`].
+///
+/// The layout is as follows:
+///
+/// ```text
+/// [account_id_second_felt (56 bits) | note_type (2 bits) | note_execution_hint_tag (6 bits)]
```

Comment:
I agree, but it is also could be helpful to mention that `sender_id_lo` here is a second felt of the account id (and in general that by `sender`  here we mean an `account`, which wasn't obvious for me).

Reply:
Added both, thank you!
Added a note to say that `sender_id_lo` = `sender_id_second_felt`.

I kept `sender_id_second_felt` because in Rust we always call it "second felt" while in MASM we call it "sender_id_lo". That difference in naming clashes here which is unfortunate, but not sure there's a better way to resolve it.